pkg/controller/deployment/sync.go

```diff
@@ -510,15 +510,26 @@ func (dc *DeploymentController) updateDeploymentStatus(allRSs []*extensions.Repl
 
 // isScalingEvent checks whether the provided deployment has been updated with a scaling event
 // by looking at the desired-replicas annotation in the active replica sets of the deployment.
-func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) bool {
+func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) (bool, error) {
 	newRS, oldRSs, err := dc.getAllReplicaSetsAndSyncRevision(d, false)
 	if err != nil {
-		return false
+		return false, err
 	}
 	// If there is no new replica set matching this deployment and the deployment isn't paused
 	// then there is a new rollout that waits to happen
 	if newRS == nil && !d.Spec.Paused {
```

Comment:
@bgrant0607 agreed, thanks for the explanation.

@pwittrock scaling+rolling out at the same time or in two separate quick updates that cannot be distinguished is not a common case I believe (unless what users really want is a different strategy) and even if we exposed proportional scaling as an API I don't see why it would be for Rolling deployments and not a new strategy (sounds a lot like AB deployments which is a variation of Rolling - I have been meaning to write a proposal for an AB strategy but haven't found any time yet). In my opinion, for now (and for Rolling deployments in general) the only thing we can do is document the behavior.